### DISCUSSION TITLE
Is initial_prompt supported for Whisper?

### DISCUSSION THREAD

**Discussion by vibl** [deepgram/community#179] [General help] [answered]
Is initial_prompt supported for Whisper? If not, would that be possible to tell us why?

(See https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/transcribe.py#L96)

It would compensate the lack of some important Deepgram features in Whisper (particularly keywords, which a lot of people need).

**Comment by jjmaldonis** [comment, marked as answer]
Hey @vibl! Currently an initial prompt for Whisper is not supported. I can check in with our team to understand more about that.

In the meantime, is there something about the `keywords` feature that is or isn't working for you? You're definitely right that a lot of people need it, so we're always interested in hearing about use cases and what is or isn't working for you. Or is there something about the non-whisper models (which support keywords) that is not fitting your use case?

**Comment by vibl** [comment]
Hi @jjmaldonis :-) They `keywords` feature is definitely working well for me. I was just pointing out that it isn't available for Deepgram's Whisper service and that `initial_prompt` would compensate for that.

The main reason I would prefer to use Whisper is that I use both French and English. As Nova is not available for French, I can only use Enhanced, which is more expensive.

Other (less important) reasons for using Whisper are:
- I would rather use an open-source model (less vendor lock-in, learning about the model would be a safer long term investment, etc.),
- I may need some features only offered by Whisper in the future (such as a customized initial prompt beyond just keywords).

**Reply by vibl** [reply]
Also, Whisper have less transcription errors than the Enhanced model for French.

**Reply by jjmaldonis** [reply]
Hey @vibl thanks for letting us know. We appreciate the feedback and I'll make sure to pass all this along to the team.

**Comment by mosnicholas** [comment]
@jjmaldonis -- pretty sure keywords isn't supported for the whisper models. Am I incorrect? Would love to see that being supported too, it's currently blocking me from using Deepgram directly.

**Reply by jjmaldonis** [reply]
That's right, keywords aren't supported for the whisper models but they are supported for the base, enhanced, and nova models. Is there something that is pushing you to use whisper instead of one of the other models that supports keywords?

**Reply by mosnicholas** [reply]
Whisper has frankly performed the best on our data set. It does a much better job overall with proper nouns, which is really important for my use case (eg. Nova transcribes "Capterra" as "cap terror").

**Reply by mosnicholas** [reply]
Is that on the roadmap for you?

**Reply by jjmaldonis** [reply]
Hey @mosnicholas, keywords for whisper is not on the roadmap at the moment but we are looking into it. Better support for proper nouns is something we are always improving. Just last month we released an updated version of our Nova model with better support for proper nouns, and every new model will improve this functionality. But an audio model cannot know the correct spelling for all proper nouns, which is where a feature like keywords (or whisper's prompt) is particularly useful. I'll make a note to update this thread if keywords are supported for whisper.

**Reply by mosnicholas** [reply]
Thank you @jjmaldonis. Can I ask why it is not on the roadmap, and if it is a technical limitation that's holding you back? From my perspective, Whisper is just as powerful as Nova, and deserves as much investment and capabilities as Nova.

**Reply by jjmaldonis** [reply]
Unfortunately I'm not able to answer that question the way you want it answered, but there are three things we can do:
1. First, I can say that Whisper is not as scalable as Nova. You have likely noticed that running Whisper, particularly Whisper large, is slow. Our engineering efforts have made dramatic improvements to Whisper's scalability, but at the end of the day it just isn't as scalable as Nova. Whisper also doesn't support streaming, which is a critical use case. I can't say much more than this, but maybe you can draw some conclusions/hypotheses about the things you're asking. -- And take this paragraph with a grain of salt, it's not meant to mean more than what it says.
2. It sounds like you're interested in confirming that Deepgram is working towards solutions that support your use cases and needs. It's likely worth your time to chat with our customer-facing teams to ensure we're building the technology you're interested in. Plus, you may be able to glean more information during those calls.
3. Lastly, can you "help me help you" by providing more information about your use case? The more targeted information I can pass to our product and engineering teams, the more helpful this feedback will be. We thrive when we get detailed customer feedback, so additional info from you can go a long way (even if you can't see it externally). So here are a handful of questions:
  - What language(s) are you transcribing? (e.g. English, Spanish, and German)
  - What kinds of proper nouns are causing the most issues? Names of companies? Names of locations? Names of people?
  - How many proper nouns do you need to transcribe? An order of magnitude estimate is helpful. For example, some of our customers need 1-10 proper nouns to be transcribed correctly, whereas other customers need high accuracy for 50,000+ proper nouns.
  - Are you currently using a work-around to fix transcription issues with proper nouns? What is or isn't working for you, and/or what do you think will work better for your use case?
  - Are there specific audio conditions that result in good or bad identification of proper nouns? For example, does your audio have a low/medium/high amount of background noise?
  - Are there other qualities of the audio files that make them unique, such as a short duration or speakers with strong accents?

This type of additional info will help us make the improvements you're looking for.

**Reply by mosnicholas** [reply]
@jjmaldonis -- happy to schedule some time if it would be easier / faster to go over all this, and or send you it all over email.

**Reply by jjmaldonis** [reply]
Sounds good, I'll reach out.

**Comment by ramild** [comment]
I want to second this! Really need that initial "prompt" argument for the whisper model.

The prompt is widely important not only for the keywords, but also for other reasons. For example, if I want the model to detect some filler sounds or repeating words that the model usually ignores.

**Reply by ramild** [reply]
Hey @jjmaldonis, are you planning to add support for the initial_prompt parameter soon? It seems like a quick fix and could help with problems like keywords or the use case I mentioned earlier.

**Reply by jjmaldonis** [reply]
Hey @ramild, this is not on our roadmap at the moment but I will pass your request along to the team!

**Comment by peldszus** [comment]
I'd like to second this as well!

As long as Nova models are not available for most languages (in my case German), Whisper-large is the model to pick. However, many applications require you to tune the output to some degree. As long as neither the `keywords` feature is available for Whisper nor the `prompt` can be customized, you are missing out potential customers who want to use those models in production.

**Reply by jjmaldonis** [reply]
Hey @peldszus, the Nova 2 German model is coming soon! If you're not subscribed to Deepgram's marketing emails, you may want to sign up so you get the update when it's available.

**Reply by peldszus** [reply]
I see, glad to hear that. For all other languages one of the two features is highly desired nonetheless. But having Nova 2 German, could solve my current problem, thanks!

Any indication, how soon "soon" is? :)
